feat: add deployments

hpkoh opened the pull request.

hpkoh committed:
feat: add deployments

andrewbaruch commented on docker-compose.yml:
```diff
@@ -8,7 +8,7 @@ x-common-variables: &common-variables
   JWT_SECRET: EXAMPLE_SECRERT # pass around JWT secret so each service can verify user TODO: add to github secrets on deployment
   REFRESH_COOKIE_KEY: EXAMPLE_PEERPREPREFRESHTOKEN
   ACCESS_COOKIE_KEY: EXAMPLE_PEERPREPACCESSTOKEN
```
Quick check for cookies and secrets, are they the same for all services? If so we can consider creating a shared secret and shared configmap instead of copying to all configmap perhaps?

hpkoh commented on kubernetes/global-config.yaml:
@andrewbaruch yup we plan to have a shared config map for shared ones but i was deploying deployments 1 by 1 to check at first. But yeah thanks for pointing it out i think i missed quite a few!

hpkoh committed:
Merge branch 'master' into feat/deployment

hpkoh committed:
feat: update global conf

hpkoh committed:
feat: update kubernetes

hpkoh committed:
Merge branch 'master' into feat/deployment

hpkoh committed:
feat: update kub

hpkoh committed:
Merge branch 'master' into feat/deployment

hpkoh committed:
feat: update yaml

hpkoh committed:
feat: update ingress

didymusne commented on kubernetes/history/history-config.yaml:
```diff
@@ -0,0 +1,7 @@
+  name: history-config
+data:
+  GOOGLE_APPLICATION_CREDENTIALS: "/home/microservices/history/key.json"
+  COLLAB_SERVICE_ENDPOINT: "http://nginx:80/collaboration"
```
Missing these:
```
MONGO_DBNAME: "test"
MONGO_HISTORY: "mongodb://user:password@mongo-history:27017"
```

didymusne commented on kubernetes/document/document-config.yaml:
```diff
@@ -0,0 +1,11 @@
+data:
+  COLLAB_SERVICE_ENDPOINT: "http://nginx:80/collaboration"
+  GOOGLE_APPLICATION_CREDENTIALS: "/home/microservices/document/key.json"
+  REDIS_DOCUMENT_SYNC_HOST: "redis-document"
```
can remove these

only need
```
COLLAB_SERVICE_ENDPOINT: "http://nginx:80/collaboration"
GOOGLE_APPLICATION_CREDENTIALS: "/home/microservices/document/key.json"
REDIS_DOCUMENT_URL: redis://redis-document:6379
```

hpkoh committed:
feat: update config

hpkoh committed:
feat: update config

yongler commented on kubernetes/matching/matching-config.yaml:
the last 2 should be

````
MATCHING_REQUEST_REDIS_URL=rediss://default:e840999abd7046a39bf40c6aaf69faf9@quality-llama-43887.upstash.io:43887
MATCHING_PAIR_REDIS_URL=rediss://default:9ed22128c2b945b78ff7fd840679a80a@sought-meerkat-43858.upstash.io:43858
```

hpkoh committed:
Merge branch 'master' into feat/deployment

hpkoh committed:
feat: update deployments

didymusne commented on kubernetes/client/client-config.yaml:
```diff
@@ -0,0 +1,7 @@
+  name: client-config
+data:
+  NEXT_PUBLIC_HOST_API_KEY: "http://localhost:3000/"
+  NEXT_PUBLIC_DEBUG: "false"
```
Need one more

```
NEXT_PUBLIC_SHOULD_AUTH_ROOM=true
```

hpkoh committed:
feat: update configs

hpkoh committed:
feat: update config

hpkoh committed:
feat: update config

hpkoh committed:
feat: update collab

hpkoh committed:
feat: fix bug

hpkoh committed:
Merge branch 'master' into feat/deployment

hpkoh committed:
feat: update conf

hpkoh committed:
feat: collab

andrewbaruch committed:
Update Deployment

Removed ingress classname
Update matching with non-standard health probe
Add new env variable

hpkoh committed:
feat: update config
